tools/statplot.py

```diff
@@ -237,50 +272,74 @@ def PlotData(self, action, type, xscale, yscale):
         self.fig.canvas.draw()
         self.fig.canvas.flush_events()
 
-    def PopulateCombo(self, action, prevIterX=None, prevIterY=None):
-        for entry in sorted(self.entries.Paths()):
-            self.xCombo.append_text(entry)
-            self.yCombo.append_text(entry)
-        if action == 'load':
-            if 'ElapsedTime' in self.entries.Paths():
-                iterX = self.xCombo.get_model(). \
-                    get_iter(sorted(self.entries.Paths()).index('ElapsedTime'))
-                self.xCombo.set_active_iter(iterX)
-                self.yCombo.set_active(0)
-            else:
-                self.xCombo.set_active(0)
-                self.yCombo.set_active(1)
-        elif action == 'reload':
-            self.xCombo.set_active_iter(prevIterX)
-            self.yCombo.set_active_iter(prevIterY)
+    def ReadData(self):
+        def GatherEntries(stat2read):
+            print('Reading ' + stat2read)
+            entries = numpy.array([])
+            with open(stat2read, 'r') as fid:
+                for num, line in enumerate(fid):
+                    if line.startswith('<field'):
+                        info = etree.fromstring(line)
+                        if info.get('statistic') == 'value':
+                            entries = numpy.append(entries, info.get('name'))
+                        elif all([x in info.keys() for x in
+                                  ['components', 'material_phase']]):
+                            for i in range(int(info.get('components'))):
+                                entries = numpy.append(
+                                    entries, info.get('material_phase') + '%'
+                                    + info.get('name') + '%'
+                                    + info.get('statistic') + f'%{i}')
```

Comment:
I think this (and the following few times this is done) would be clearer as a full format strings (using the other suggestion above). Maybe something like:

```python
for i in range(...):
  entries.append("{e[material_phase]}%{e[name]}%{e[statistic]}%i".format(e=info.attrib, i=i))
```

Reply:
Cheers, looks cleaner that way!